src/MSBuildBinLogQuery/Component/BuildComponent.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.Build.Logging.Query.Messaging;
+
+namespace Microsoft.Build.Logging.Query.Component
+{
+    public abstract class BuildComponent
+    {
+        public IList<Message> Messages => _messages.ToImmutableList();
```

Comment:
You can get this effect (callers shouldn't mutate the internal list) more cheaply like this:
```suggestion
        public IReadOnlyList<Message> Messages => _messages;
```